include/crm/pengine/pe_types.h

```diff
@@ -185,6 +186,8 @@ struct pe_working_set_s {
     int ninstances;     // Total number of resource instances
     guint shutdown_lock;// How long (seconds) to lock resources to shutdown node
     int priority_fencing_delay; // Priority fencing delay
+
+    pcmk__output_t *out;
```

Comment:
I just realized we can't do this. pe_types.h is public API, we can't include an internal header or use internal types. You could make it a void* and cast it when using it.

Reply:
That does raise the question of how an external caller would use the functions that output. I think we can leave that unanswered for now -- I doubt there's demand for it. I think the new high-level APIs are more of what other developers would be interested in.

Reply:
> I just realized we can't do this. pe_types.h is public API, we can't include an internal header or use internal types. You could make it a void* and cast it when using it.

If we're going to have a void * there, it might make sense to make it a pointer to some pe_working_set_private struct that could potentially hold other things in the future.  I've been wondering if it actually makes sense to have three different pcmk__output_t structures registered on the data set - one for text/xml/whatever, one for logging, and one for none - and then just change which one is active as needed.

Reply:
The nice thing about void* is we can make it the output object now and expand it to a struct if needed later. Worth calling it void *private I suppose.
I don't think we want the overhead of multiple output objects in that. That logging instance is an oddball, I'm not sure if/how it should be generalized. But even if we do, it would still only be one output object per caller -- pacemaker-schedulerd would use the log formatter, while crm_simulate would use its command-line one.